drivers/rail/rail_netdev.c

```diff
@@ -0,0 +1,856 @@
+            ret = sizeof(uint8_t);
+            break;
+
+        case (NETOPT_BANDWIDTH):
+            /*
+                   assert(max_len >= sizeof(int8_t));
+                   // bits/seconds
+                   uint32_t bw = RAIL_BitRateGet();
+                   uint8_t bw_kb = bw/1000;
+             *((uint8_t *)val) = bw_kb;
+                   ret = sizeof(uint8_t);
+             */
+            break;
+        default:
+            /* DEBUG("not supported netopt code at rail drv %d str %s \n", opt,
+               netopt2str(opt)); */
+            break;
+    }
+    if (ret != -ENOTSUP) {
+        return ret;
+    }
+
+    ret = netdev_ieee802154_get((netdev_ieee802154_t *)netdev, opt, val, max_len);
+
+    if (ret != -ENOTSUP) {
+        return ret;
+    }
+
+    /*DEBUG("ieee802.15.4 could not handle netopt opt %s \n", netopt2str(opt));*/
+
+    return ret;
+}
+
+static int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t len)
+{
+
+
+    if (netdev == NULL) {
+        return -ENODEV;
+    }
+
+    rail_t *dev = (rail_t *)netdev;
+    int res = -ENOTSUP;
+    RAIL_Status_t rail_ret;
+    uint16_t le_u16;
+    uint64_t le_u64;
+
+    /* TODO wake up transceiver? necessary? or done automaticaly by RAIL driver
+       blob?
+     */
+
+    switch (opt) {
+
+        case (NETOPT_CHANNEL):
+            /* since we have to provide the channel for each tx or rx, just
+               change the attribute in the netdev struct */
+            assert(len == sizeof(uint16_t));
+            uint8_t chan = (((const uint16_t *)val)[0]) & UINT8_MAX;
+
+            if (dev->params.freq == RAIL_TRANSCEIVER_FREQUENCY_2P4GHZ) {
+                if (chan < RAIL_2P4GH_MIN_CHANNEL || chan > RAIL_2P4GH_MAX_CHANNEL) {
+                    res = -EINVAL;
+                    break;
+                }
+            }
+            else if (dev->params.freq == RAIL_TRANSCEIVER_FREQUENCY_868MHZ) {
+                /* 868MHz has only one channel, channel 0! */
+                if (chan != RAIL_868MHZ_DEFAULT_CHANNEL) {
+                    res = -EINVAL;
+                    break;
+                }
+            }
+            else if (dev->params.freq == RAIL_TRANSCEIVER_FREQUENCY_912MHZ) {
+                if (chan > RAIL_912MHZ_MAX_CHANNEL) {
+                    res = -EINVAL;
+                    break;
+                }
+            }
+            else {
+                res = -EINVAL;
+                LOG_ERROR("Unknown radio frequency configured\n");
+                assert(false);
+                break;
+            }
+            /* since we have to provide the channel for each tx or rx, just
+               change the attribute in the netdev struct */
+            /* TODO if chan_old != chan_new -> interupt rx op? */
+            /* don't set res to set netdev_ieee802154_t::chan */
+            dev->netdev.chan = chan;
+
+            break;
+        case (NETOPT_CHANNEL_PAGE):
+            /* TODO?? */
```

Comment:
Maybe we asses if this is a TODO or not before we merge ;-)

Reply:
ok, done :)